apps/api_web/lib/api_web/controllers/route_controller.ex

```diff
@@ -139,8 +139,13 @@ defmodule ApiWeb.RouteController do
   defp do_filter(%{stops: _stops, type: types} = filters) do
     filters
     |> routes_at_stops()
-    |> Enum.flat_map(fn id -> for type <- types, do: %{id: id, type: type} end)
-    |> Route.select()
+    |> routes_by_id_and_type(types)
+  end
+
+  defp do_filter(%{service_ids: [], type: _}), do: []
+
+  defp do_filter(%{service_ids: [service_id | _], type: types}) do
+    service_id |> RoutesByService.for_service_id() |> routes_by_id_and_type(types)
```

Comment:
Since the `type` is a route type, could you also incorporate that into what `RoutesByService` lets you query? Something like `for_service_id_and_type()`?